src/rgw/rgw_sal_motr.cc

```diff
@@ -366,6 +366,43 @@ int MotrUser::read_stats(const DoutPrefixProvider *dpp,
     ceph::real_time *last_stats_sync,
     ceph::real_time *last_stats_update)
 {
+  int rc, max_entries = 100; // to fetch in chunks of 100
+  vector<string> keys(max_entries);
+  vector<bufferlist> vals(max_entries);
+  std::string user_stats_iname = "motr.rgw.user.stats." + info.user_id.to_str();
+  rgw_bucket_dir_header bkt_header;
+
+  do
+  {
+      rc = store->next_query_by_name(user_stats_iname, keys, vals);
+      if (rc < 0) {
+          ldpp_dout(dpp, 20) << __func__ << ": failed to get the user stats info for user  = "
+                            << info.user_id.to_str() << dendl;
+          return rc;
+      }
+
+    for (int i = 0 ; i < rc; i++)
+    {
+        ceph::buffer::list::const_iterator bitr = vals[i].begin();
+        bkt_header.decode(bitr);
+
+        for (const auto& pair : bkt_header.stats) 
+        {
+          const rgw_bucket_category_stats& header_stats = pair.second;
+
+          stats->num_objects += header_stats.num_entries;
+          stats->size += header_stats.total_size;
+          stats->size_utilized += header_stats.actual_size;
```

Comment:
as we have decided not to support yet the size_actual & the size_utilized yet.
Still to honor the common code, we should set size_rounded.
```suggestion
          stats->size_ rounded += rgw_rounded_kb(header_stats.actual_size) * 1024;
```

Reply:
Addressed